x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/analytics/process/AnalyticsControlMessageWriter.java

```diff
@@ -18,7 +18,7 @@
      * but in the context of the java side it is more descriptive to call this the
      * end of data message.
      */
-    private static final String END_OF_DATA_MESSAGE_CODE = "r";
+    private static final String END_OF_DATA_MESSAGE_CODE = "$";
```

Comment:
I think it's still `'r'` on the C++ side.

Reply:
This https://github.com/elastic/ml-cpp/pull/334/files changes it to `$`